blackhole.c

```diff
@@ -714,6 +776,12 @@ static int blackhole_configure_outbound_atu(struct tenstorrent_device *tt_dev, u
 	return 0;
 }
 
+static void blackhole_create_sysfs_groups(struct tenstorrent_device *tt_dev) {
+	int ret = devm_device_add_group(&tt_dev->dev, &bh_pcie_perf_counters_group);
```

Comment:
These are added to `tt_dev->dev`, is that the same kobj as `/dev/tenstorrent/n`? So these files will end up in `/sys/bus/pci/drivers/tenstorrent/0000:01:00.0/tenstorrent/tenstorrent!n`? 

(That ! in the path is kind of awkward)

The other option is you could hang it off the hwmon device using the extra_groups argument to devm_hwmon_device_register_with_info. This is not considered great practice by the hwmon maintainer, but I thought I'd mention it.

edit: I see you described all this in your docs patch. Ignore my comments.